libs/core/thread_pools/include/hpx/thread_pools/scheduled_thread_pool_impl.hpp

```diff
@@ -194,7 +194,13 @@ namespace hpx { namespace threads { namespace detail {
                 hpx_thread_offset;
         bool have_polling_work =
             sched_->Scheduler::get_polling_work_count() > 0;
-        return have_hpx_threads || have_polling_work;
+
+        if (have_hpx_threads || have_polling_work)
+        {
+            sched_->Scheduler::cleanup_terminated(true);
+            return true;
+        }
+        return false;
```

Comment:
Is this essential for fixing whatever shutdown problems this is fixing? Note that `threadmanager::cleanup_terminated` is called inside the `yield_while` loops in `runtime_support_server.cpp`.

Reply:
It was not necessary, but helped speeding up the shutdown process as terminated threads were cleaned up right away.

Reply:
I guess I just don't see how this helps speed things up. Is there a place where `threadmanager::is_busy` is called repeatedly without a corresponding `threadmanager::cleanup_terminated` right next to it? I think I'd prefer it if `is_busy` really did just check if the pool is busy or not.

Reply:
Yes, there is a place where this happens. I'll go back and change things there.

Reply:
I was wrong, everything is ok here. I have reverted this change.